webpack/assets/javascripts/react_app/components/Layout/components/TaxonomyDropdown.js

```diff
@@ -4,62 +4,98 @@ import NavItem from './NavItem';
 import { noop } from '../../../common/helpers';
 import { translate as __ } from '../../../common/I18n';
 
-const TaxonomyDropdown = ({
-  taxonomyType,
-  currentTaxonomy,
-  taxonomies,
-  id,
-  changeTaxonomy,
-  anyTaxonomyText,
-  manageTaxonomyText,
-  anyTaxonomyURL,
-  manageTaxonomyURL,
-}) => (
-  <NavItem className="dropdown org-switcher" id={id}>
-    <a
-      href="#"
-      className="dropdown-toggle nav-item-iconic"
-      data-toggle="dropdown"
-    >
-      {__(currentTaxonomy)}
-      <span className="caret" />
-    </a>
-    <ul className="dropdown-menu">
-      <li className="dropdown-header">{__(taxonomyType)}</li>
-      <li>
+class TaxonomyDropdown extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { searchValue: '' };
+  }
+
+  render() {
+    const {
+      taxonomyType,
+      currentTaxonomy,
+      taxonomies,
+      id,
+      changeTaxonomy,
+      anyTaxonomyText,
+      manageTaxonomyText,
+      anyTaxonomyURL,
+      manageTaxonomyURL,
+    } = this.props;
+
+    const filteredTaxonomies = () => {
+      const { searchValue } = this.state;
+
+      if (searchValue === '') {
+        return this.props.taxonomies;
```

Comment:
nit: you already use destructuring above to get taxonomies from props so you can take advantage of that and pass them as an arg to your filter function - that way you do not need to pull taxonomies out of the props here:

```js
const filteredTaxonomies = taxonomiesProp => {
//
}

// on l.80
filteredTaxonomies(taxonomies).map
```